### PR TITLE
Add rule sshd_enable_pubkey_auth

### PR DESCRIPTION
**Pull request by dodys**
#### Description:

- As part of UBTU-20-010033 (Ubuntu 20.04 STIG):

Configure the Ubuntu operating system to use multifactor authentication for network access to accounts.
1. Add or update `pam_pkcs11.so` in "/etc/pam.d/common-auth" to match the following line:
`auth [success=2 default=ignore] pam_pkcs11.so`
2. Set the sshd option `PubkeyAuthentication yes` in the "/etc/ssh/sshd_config" file.

1 is already achieved by smartcard_pam_enabled, so we added the missing second step.

### CODE REVIEW

File: linux_os/guide/services/ssh/ssh_server/sshd_enable_pubkey_auth/rule.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
+documentation_complete: true
+
+title: 'Enable PubkeyAuthentication'
+
+description: |-
+    Verify the sshd daemon allows public key authentication with the
+    following:
+    <pre>$ grep ^Pubkeyauthentication /etc/ssh/sshd_config
```
Comment (jan-cerny) [inline linux_os/guide/services/ssh/ssh_server/sshd_enable_pubkey_auth/rule.yml:8]:
```suggestion
    <pre>$ grep ^PubkeyAuthentication /etc/ssh/sshd_config
```

File: linux_os/guide/services/ssh/ssh_server/sshd_enable_pubkey_auth/rule.yml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    authentication.
+
+severity: medium
+
+references:
+    disa: CCI-000765,CCI-000766,CCI-000767,CCI-000768
+    srg: SRG-OS-000105-GPOS-00052,SRG-OS-000106-GPOS-00053,SRG-OS-000107-GPOS-00054,SRG-OS-000108-GPOS-00055
+    stigid@ubuntu2004: UBTU-20-010033
+
+ocil_clause: 'it is not disable'
```
Comment (jan-cerny) [inline linux_os/guide/services/ssh/ssh_server/sshd_enable_pubkey_auth/rule.yml:27]:
```suggestion
ocil_clause: 'it is not enabled'
```

File: linux_os/guide/services/ssh/ssh_server/sshd_enable_pubkey_auth/rule.yml [hunk 3]
```diff
@@ -0,0 +1,41 @@
+documentation_complete: true
+
+title: 'Enable PubkeyAuthentication'
```
Comment (jan-cerny) [inline linux_os/guide/services/ssh/ssh_server/sshd_enable_pubkey_auth/rule.yml:3]:
```suggestion
title: 'Enable Public Key Authentication'
```

**Review by jan-cerny — COMMENTED** [review]
Looks great, I have only few nit picks.

File: linux_os/guide/services/ssh/ssh_server/sshd_enable_pubkey_auth/rule.yml (hunk 1 above)
Comment (dodys) [reply linux_os/guide/services/ssh/ssh_server/sshd_enable_pubkey_auth/rule.yml]: thanks for the suggestions, I've added it and rebased.

**Review by jan-cerny — APPROVED** [review]

**Pull request merged**